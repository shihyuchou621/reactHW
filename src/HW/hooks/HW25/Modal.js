import cx from 'classnames';

import React from 'react';

import style from './style.module.css';

export default function Modal(props) {
  const {
    header,
    position,
    isAnimation,
    children,
    onClose,
    closeText,
    btnClass,
  } = props;

  return (
    <div className={style.modalBg}>
      <div className={cx(style.allModal, {
        [style.myModal]: position === "center",
        [style.myTop]: position === "top",
        [style.myAnimation]: isAnimation,
        [style.myAnimation1]: isAnimation,
      })}>
        {header &&
          <h2>{header}</h2>
        }
        <div data-testid="content">
          {children}
        </div>
        <button
          className={btnClass}
          onClick={onClose}
        >
          {closeText}
        </button>
      </div>
    </div>
  );
}
