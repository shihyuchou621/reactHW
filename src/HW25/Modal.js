import cx from 'classnames';

import React, { Component } from 'react'

export default class Modal extends Component {   
  render() {
    const { 
      header,
      position,
      isAnimation,
    } = this.props;

    return (
      <div className="modalBg">
        <div className={cx("allModal", {
          myModal: position === "center",
          myTop: position === "top",
          myAnimation: isAnimation,
          myAnimation1: isAnimation,
        })}>
          {header &&
          <h2>{header}</h2>
          }
          <div>{this.props.children}</div>
          <button 
            className="btn btn-primary btn-block"
            onClick={this.props.onClose}
          >
            關閉
          </button>
        </div>
      </div>
    )
  }
}