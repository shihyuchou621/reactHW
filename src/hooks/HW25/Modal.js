import cx from 'classnames';

import React from 'react';

export default function Modal(props) {
  const {
    header,
    position,
    isAnimation,
    children,
    onClose
  } = props;

  return (
    <div className="modalBg">
      <div className={cx("allModal", {
        myModal: position === "center",
        myTop: position === "top",
        myAnimation: isAnimation,
        myAnimation1: isAnimation,
      })}>
        {header &&
          <h2>{header}</h2>
        }
        <div data-testid="content">
          {children}
        </div>
        <button
          className="btn btn-primary btn-block"
          onClick={onClose}
        >
            關閉
        </button>
      </div>
    </div>
  );
}
