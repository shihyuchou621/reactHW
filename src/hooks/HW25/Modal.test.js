import React from 'react';

import {
  cleanup,
  render,
  fireEvent,
} from '@testing-library/react';

import Modal from './Modal';

afterEach(cleanup);

test('Modal', () => {
  const onClose = jest.fn();
  const { container, getByText, getByTestId } = render(
    <Modal
      name="Modal"
      position="center"
      onClose={onClose}
    >lorem</Modal>
  );

  fireEvent.click(getByText("關閉"));
  expect(onClose.mock.calls.length).toBe(1);
  expect(getByTestId("content").innerHTML).toBe("lorem");
  expect(container).toMatchSnapshot();
});

test('top', () => {
  const onClose = jest.fn();
  const { container, getByText, getByTestId } = render(
    <Modal
      name="top"
      position="top"
      onClose={onClose}
    >lorem</Modal>
  );

  fireEvent.click(getByText("關閉"));
  expect(onClose.mock.calls.length).toBe(1);
  expect(getByTestId("content").innerHTML).toBe("lorem");
  expect(container).toMatchSnapshot();
});

test('header', () => {
  const onClose = jest.fn();
  const { container, getByText, getByTestId } = render(
    <Modal
      name="header"
      position="center"
      onClose={onClose}
    >lorem</Modal>
  );

  fireEvent.click(getByText("關閉"));
  expect(onClose.mock.calls.length).toBe(1);
  expect(getByTestId("content").innerHTML).toBe("lorem");
  expect(container).toMatchSnapshot();
});

test('top + animation', () => {
  const onClose = jest.fn();
  const { container, getByText, getByTestId } = render(
    <Modal
      name="top + animation"
      isAnimation={true}
      onClose={onClose}
    >lorem</Modal>
  );

  fireEvent.click(getByText("關閉"));
  expect(onClose.mock.calls.length).toBe(1);
  expect(getByTestId("content").innerHTML).toBe("lorem");
  expect(container).toMatchSnapshot();
});
