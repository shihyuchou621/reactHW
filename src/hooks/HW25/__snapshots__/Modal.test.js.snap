// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal 1`] = `
<div>
  <div
    class="modalBg"
  >
    <div
      class="allModal myModal"
    >
      <div
        data-testid="content"
      >
        lorem
      </div>
      <button
        class="btn btn-primary btn-block"
      >
        關閉
      </button>
    </div>
  </div>
</div>
`;

exports[`header 1`] = `
<div>
  <div
    class="modalBg"
  >
    <div
      class="allModal myModal"
    >
      <div
        data-testid="content"
      >
        lorem
      </div>
      <button
        class="btn btn-primary btn-block"
      >
        關閉
      </button>
    </div>
  </div>
</div>
`;

exports[`top + animation 1`] = `
<div>
  <div
    class="modalBg"
  >
    <div
      class="allModal myAnimation myAnimation1"
    >
      <div
        data-testid="content"
      >
        lorem
      </div>
      <button
        class="btn btn-primary btn-block"
      >
        關閉
      </button>
    </div>
  </div>
</div>
`;

exports[`top 1`] = `
<div>
  <div
    class="modalBg"
  >
    <div
      class="allModal myTop"
    >
      <div
        data-testid="content"
      >
        lorem
      </div>
      <button
        class="btn btn-primary btn-block"
      >
        關閉
      </button>
    </div>
  </div>
</div>
`;
